app/src/main/java/org/mifos/mobile/ui/beneficiary/presentation/BeneficiaryScreen.kt

```diff
@@ -0,0 +1,90 @@
+package org.mifos.mobile.ui.beneficiary.presentation
+
+import android.content.res.Configuration
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import org.mifos.mobile.R
+import org.mifos.mobile.core.ui.component.MifosTopBar
+import org.mifos.mobile.core.ui.theme.MifosMobileTheme
+
+
+/**
+ * This is a composable function that represent the beneficiary screen and the parameters include
+ * @param[addiconClicked] to navigate to the add screen
+ * @param[scaniconClicked] to navigate to the scanicon screen
+ * @param[uploadiconClicked] to navigate to the upload icin screen
+ **/
+
+
+@Composable
+fun BeneficiaryScreen(
+    topAppbarNavigateback: () -> Unit,
+    addiconClicked: () -> Unit,
+    scaniconClicked: () -> Unit,
+    uploadiconClicked: () -> Unit
+) {
+    MifosTopBar(
+        navigateBack = topAppbarNavigateback,
+        title = {
+            Text(text = stringResource(id = R.string.add_beneficiary))
+        }
+    )
+
+    Column(
+        modifier = Modifier.padding(dimensionResource(id = R.dimen.padding_10dp))
+    ) {
+        Text(
+            stringResource(id = R.string.select_mode),
+            fontSize = 18.sp,
+            color = MaterialTheme.colorScheme.onSurface
+        )
+
+        Text(
+            modifier = Modifier.padding(top = 16.dp),
+            text = stringResource(R.string.add_beneficiary_option),
+            fontSize = 18.sp,
+            color = MaterialTheme.colorScheme.onSurface
+        )
+
+        BeneficiaryScreenIcons(
+            modifier = Modifier.padding(top = 20.dp),
+            addIconclicked = addiconClicked,
+            scanIconClicked = scaniconClicked,
+            uploadIconClicked = uploadiconClicked
+        )
+
+    }
+
+}
+
+@Preview(
+    name = "Night Mode",
+    uiMode = Configuration.UI_MODE_NIGHT_YES,
+)
+@Preview(
+    name = "Day mode",
+    uiMode = Configuration.UI_MODE_NIGHT_NO,
+)
+@Composable
+fun BeneficiaryScreenPreview() {
+    MifosMobileTheme {
+        Column(
```

Comment:
@UduakUmanah10 Preview should have anything extra like column etc, It is meant to see what exactly will be visible on screen. Please remove from everywhere

Reply:
![Screenshot (127)](https://github.com/openMF/mifos-mobile/assets/90254566/2602b7e1-a035-4396-8062-7254409c2b7f)